polaris-react/src/components/Modal/components/Dialog/Dialog.scss

```diff
@@ -68,9 +66,7 @@ $breakpoints-height-limit-up: breakpoints-up($height-limit + $vertical-spacing);
 
   &.limitHeight {
     @media #{$p-breakpoints-md-up} {
-      @media #{$breakpoints-height-limit-up} {
```

Comment:
I would be replacing this media condition with `@media #{$p-breakpoints-md-up} {` which would duplicate the media condition above it so I just removed it.